src/test/java/guitests/BoardAutoCreatorTest.java

```diff
@@ -105,14 +104,22 @@ public void sampleBoardAutoCreationTest() {
         waitUntilNodeAppears(SAMPLE_BOARD_DIALOG);
         click("OK");
 
-        assertEquals(panelControl.getPanelCount(), SAMPLE_PANEL_NAMES.size());
+        assertEquals(panelControl.getPanelCount(), BoardAutoCreator.getSamplePanelDetails().size());
         assertEquals(panelControl.getCurrentlySelectedPanel(), Optional.of(0));
         assertEquals(panelControl.getNumberOfSavedBoards(), 1);
+        testSamplePanelInfos(panelControl);
+    }
 
-        List<PanelInfo> panelInfos = panelControl.getCurrentPanelInfos();
-        for (int i = 0; i < SAMPLE_PANEL_NAMES.size(); i++){
-            assertEquals(panelInfos.get(i).getPanelFilter(), SAMPLE_PANEL_FILTERS.get(i));
-            assertEquals(panelInfos.get(i).getPanelName(), SAMPLE_PANEL_NAMES.get(i));
+    /**
+     * Tests whether the sample panels along with their filters are rightly created and present in the UI
+     */
+    public static void testSamplePanelInfos(PanelControl pc){
+        List<PanelInfo> panelInfos = pc.getCurrentPanelInfos();
+        int i = 0;
+        for (String panelName : BoardAutoCreator.getSamplePanelDetails().keySet()) {
+            assertEquals(panelInfos.get(i).getPanelName(), panelName);
+            assertEquals(panelInfos.get(i).getPanelFilter(), BoardAutoCreator.getSamplePanelDetails().get(panelName));
+            i++;
```

Comment:
This method has high coupling because it takes more than it needs. Comparing boards doesn't need `PanelControl` or `BoardAutoCreator`. Can you do this instead? `verifySampeBoard(pc.getCurrentPanelInfos(), BoardAutoCreator.getSamplePanelDetails())`?